application/usecases/instagram_posts.go

```diff
@@ -0,0 +1,164 @@
+package usecases
+
+import (
+	"TML_TBot/config"
+	"TML_TBot/domain/models"
+	"bytes"
+	"fmt"
+	"image"
+	"image/png"
+	"net/http"
+	"os"
+	"reflect"
+
+	"github.com/Davincible/goinsta/v3"
+)
+
+type InstagramPostsController struct {
+	feedItems []FeedItem
+	codes     CodesSet
+}
+
+type FeedItem struct {
+	text  string
+	image *[]byte
+}
+
+// Use a set to avoid duplicate codes
+type CodesSet map[string]void
+
+type void struct{}
```

Comment:
Both of the above instructions denote other languages influence. Codes**Set** = **map**. 'void' seems also wanting to be what in some langs is known as void. You can use `interface{}` or now `any` to get familiar with Go patterns :)


```suggestion
type CodesMap map[string]any
```

(you would need to perform a little refactor bc of this)

Reply:
Done!